swatch-billable-usage/src/main/java/com/redhat/swatch/billable/usage/services/BillableUsageStatusConsumer.java

```diff
@@ -48,9 +55,37 @@ public void process(BillableUsageAggregate billableUsageAggregate) {
           "Billable Usage Status update received with no status, {}:", billableUsageAggregate);
       return;
     }
+
+    updateStatus(billableUsageAggregate);
+    if (billableUsageAggregate.getStatus() == FAILED
+        && billableUsageAggregate.getErrorCode() == SUBSCRIPTION_NOT_FOUND) {
+      sendToDeadLetterTopic(billableUsageAggregate);
+    }
+  }
+
+  private void sendToDeadLetterTopic(BillableUsageAggregate billableUsageAggregate) {
```

Comment:
Since this is now in the same service I think it make more sense to set the retry_after directly instead of going through the dlt.

Reply:
I like this idea, but this is out of scope for this ticket and needs to be discussed in the refinement. I've created a ticket for this: https://issues.redhat.com/browse/SWATCH-2803

Reply:
Sounds good thanks!